osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/k8s/securitygroup/CreateK8sLabelPodTask.java

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.core.broker.service.tasks.conformance.k8s.securitygroup;
+
+import javax.persistence.EntityManager;
+
+import org.apache.log4j.Logger;
+import org.osc.core.broker.model.entities.virtualization.SecurityGroup;
+import org.osc.core.broker.model.entities.virtualization.SecurityGroupMember;
+import org.osc.core.broker.model.entities.virtualization.k8s.Label;
+import org.osc.core.broker.model.entities.virtualization.k8s.Pod;
+import org.osc.core.broker.model.entities.virtualization.k8s.PodPort;
+import org.osc.core.broker.model.plugin.ApiFactoryService;
+import org.osc.core.broker.rest.client.k8s.KubernetesPod;
+import org.osc.core.broker.service.exceptions.VmidcException;
+import org.osc.core.broker.service.persistence.OSCEntityManager;
+import org.osc.core.broker.service.persistence.PodEntityMgr;
+import org.osc.core.broker.service.tasks.TransactionalTask;
+import org.osc.sdk.controller.api.SdnRedirectionApi;
+import org.osc.sdk.controller.element.NetworkElement;
+import org.osgi.service.component.annotations.Reference;
+
+public class CreateK8sLabelPodTask extends TransactionalTask{
+    private static final Logger LOG = Logger.getLogger(CreateK8sLabelPodTask.class);
+
+    private KubernetesPod k8sPod;
+    private Label label;
+
+    @Reference
+    private ApiFactoryService apiFactoryService;
+
+    @Override
+    public void executeTransaction(EntityManager em) throws Exception {
+        OSCEntityManager<Label> dsEmgr = new OSCEntityManager<Label>(Label.class, em, this.txBroadcastUtil);
+        this.label = dsEmgr.findByPrimaryKey(this.label.getId());
+
+        Pod existingPod = PodEntityMgr.findExternalId(em, this.k8sPod.getUid());
+        SecurityGroup thisSG = this.label.getSecurityGroupMembers().iterator().next().getSecurityGroup();
+
+        if (existingPod != null) {
+            for (Label existingPodLabel : existingPod.getLabels()) {
+                if (existingPodLabel.getMarkedForDeletion() == true) {
+                    continue;
+                }
+
+                for (SecurityGroupMember sgm : existingPodLabel.getSecurityGroupMembers()) {
```

Comment:
Say the sgm is not a part of thisSG. Who could have marked it for deletion. Is it still locked by that other task right now? And if not, does that mean the other task has completed without actually deleting

Reply:
It could be part of another SG that is either in the process of deletion of whose deletion failed to finish. The user could have triggered the deletion.